src/Entity/BaseGroup.php

```diff
@@ -0,0 +1,253 @@
+{
+    const APPROVED = 'APPROVED';
+    const PENDING = 'PENDING';
+    const REFUSED = 'REFUSED';
+
+    use EntityIdentityTrait;
+    use EntityCrudTrait;
+    use EntityTimestampableTrait;
+    use EntityPostAddressTrait;
+
+    /**
+     * The group name.
+     *
+     * @ORM\Column
+     *
+     * @Algolia\Attribute
+     */
+    protected $name;
+
+    /**
+     * The group name.
+     *
+     * @ORM\Column
+     *
+     * @Algolia\Attribute
+     */
+    protected $canonicalName;
+
+    /**
+     * The group slug.
+     *
+     * @ORM\Column
+     *
+     * @Gedmo\Slug(fields={"canonicalName"})
+     *
+     * @Algolia\Attribute
+     */
+    protected $slug;
+
+    /**
+     * The group description.
+     *
+     * @ORM\Column(type="text")
+     *
+     * @Algolia\Attribute
+     */
+    protected $description;
+
+    /**
+     * The group current status.
+     *
+     * @ORM\Column(length=20)
+     */
+    protected $status;
+
+    /**
+     * The timestamp when an administrator approved this group.
+     *
+     * @ORM\Column(type="datetime", nullable=true)
+     */
+    protected $approvedAt;
+
+    /**
+     * The timestamp when an administrator refused this group.
+     *
+     * @ORM\Column(type="datetime", nullable=true)
+     */
+    protected $refusedAt;
+
+    /**
+     * The adherent UUID who created this group.
+     *
+     * @ORM\Column(type="uuid", nullable=true)
+     */
+    protected $createdBy;
+
+    /**
+     * @ORM\Column(type="phone_number", nullable=true)
+     */
+    protected $phone;
+
+    /**
+     * The cached number of members (followers and hosts).
+     *
+     * @ORM\Column(type="smallint", options={"unsigned": true})
+     *
+     * @Algolia\Attribute
+     */
+    protected $membersCounts;
+
+    public function __toString()
+    {
+        return $this->name ?: '';
+    }
+
+    public static function createUuid(string $name): UuidInterface
+    {
+        return Uuid::uuid5(Uuid::NAMESPACE_OID, static::canonicalize($name));
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function getSlug(): string
+    {
+        return $this->slug;
+    }
+
+    public function getPostAddress(): PostAddress
+    {
+        return $this->postAddress;
+    }
+
+    public function getDescription(): string
+    {
+        return $this->description;
+    }
+
+    public function isWaitingForApproval(): bool
+    {
+        return self::PENDING === $this->status && !$this->approvedAt;
+    }
+
+    /**
+     * @Algolia\IndexIf
+     */
+    public function isApproved(): bool
+    {
+        return self::APPROVED === $this->status && $this->approvedAt;
+    }
+
+    public function isPending(): bool
+    {
+        return self::PENDING === $this->status;
+    }
+
+    public function isRefused(): bool
+    {
+        return self::REFUSED === $this->status;
+    }
+
+    public function getMembersCount(): int
+    {
+        return $this->membersCounts;
```

Comment:
Property not initialized as 0, getter may break.

Reply:
Initialized in constructor.